### DISCUSSION TITLE
뜬금없지만 '자가진단 매크로는 옳은가?'

### DISCUSSION THREAD

**Discussion by lhwdev** [lhwdev/covid-selftest-macro#21] [일반]
# 과연 자가진단 매크로는 옳을까?
사실 제가 이 생각을 안 해본 것도 아니고 한두번 생각해본 것도 아닙니다.
과연 건강상태 자가진단에 대한 매크로 앱을 만들어서 배포하고, 이용하는 게 옳은 행위일까요?

법적인 차원과 윤리적인 차원으로 나눠서 생각해봐야겠지만 저는 법을 잘 모르기 때문에...
## 법적 차원
리버스 엔지니어링이나 패킷 분석 등에 대해서 그렇게까지 규제가 심했던 것 같지는 않습니다. 예를 들어 반디집이 알집의 .alz 포맷을 리버스 엔지니어링하여 독자적인 압축 해제 알고리즘을 만들었던 것처럼요. (물론 그 사례가 여기에 적용되는지는 모르겠습니다)

최근이지만 개인정보 처리 방침을 만들었기도 하고, 이 앱은 제가 운영하는 서버에서 데이터를 처리하는 것처럼 중간단계를 거치는 형태가 아니라 바로 교육청의 서버로 데이터를 보내는 형태를 띕니다. 저는 개인정보 측면에서 문제가 될 것이라고 생각하지 않습니다만 실제로 어떨지는 모르겠네요. 확실한 건 저는 여러분의 데이터를 함부로 다루지 않을 겁니다.

건강상태 자가진단이라는 외부 서비스에 트래픽을 허가없이 보내는 것이기 때문에 법적으로 문제가 될 수도 있습니다. 다만 이 앱은 사용자의 휴대전화에서만 이러한 작업을 하기 때문에 사용자에게 책임이 귀속될 수도 있겠습니다. 확실한건 그저 나 자신이 편하려고 앱을 스스로 만들어서 쓴 것과, 앱을 유포하는 것은 완전히 다른 차원의 일이죠.

## 윤리적 차원
윤리적인 차원에서 이야기할 때는 여러 사람들의 의견이 갈릴 것이라고 생각합니다. 그냥 편하니까 된다고 생각하는 사람, 자가진단의 실효성 등의 이유로 상관없다고 생각하는 사람, 안된다고 생각하면서 쓰는 사람, 방역망에 혼선을 줄 수도 있다는 것 등의 이유로 안된다고 하는 사람이 있겠죠.  
우선 어느 것이 맞을지 생각해보기 전에 다른 의견을 가진 사람들에 대한 존중을 우선시해야 한다고 생각합니다. 이러한 전제를 두고 생각해봅시다.

아래의 의견들은 가상의 찬성/반대 의견일 뿐, 객관적인 사실이 전혀 아님을 밝혀드립니다. 그리고 최대한 중립적인 관점을 취하려 했지만 그렇지 못했을 수도 있다는 점 양해 부탁드립니다.

### **건강상태 자가진단의 실효성**  
- **건강상태 자가진단은 실효성이 없습니다**. 학생들이 매일 자가진단을 하다 보면 무감각해져서 기계적으로 '증상 없음'을 누르게 되고, 이 자체로 방역에 실질적인 도움이 되지 않습니다. 따라서 기계적으로 누르는 것과 매크로로 대체하는 것 둘다 실효성이 없으므로 큰 차이가 없다고 볼 수 있습니다.
- **실효성이 있습니다.** 코로나19에 감염되었는데도 자가진단 매크로 앱(이 앱인지 다른 앱인지는 몰라요)을 이용한 사례가 있었습니다. 이러한 상황에서 자가진단은 감염병의 확산을 분명히 막아줄 수 있습니다.
- **실효성이 있는지 여부는 큰 문제가 되지 않습니다**. 실제로 실효성이 없다고 확신할 수도 없고, 이러한 절차를 무시하고 매크로로 대체하는 것은 바르지 않다고 할 수 있습니다.

### **편익**
- **학생들의 편익을 높여주고 그에 비해 국가에 입히는 손실이 미미하기에 큰 문제가 되지 않습니다.** 이 대목을 뒷받침하려면 자가진단의 실효성을 다뤄야 합니다. 다만 학생들의 편익을 높여준다는 점은 분명합니다.
- **국가에 입히는 손실이 지대하기 때문에 문제가 됩니다.** '건강상태 자가진단의 실효성' 문단과 관련이 있는 듯 합니다. 공익의 손실이 더 크기 때문에 지양해야 한다는 관점입니다.

편익은 지나치게 주관적인 것이기에, 개개인의 판단에 맡겨야 할 것 같네요.
다만 제 입장에서는 이 앱을 처음 만들 때에도 제가 귀찮아서 만든 것이고, 공개적으로 배포한 이유도 다른 사람들의 편의를 위한 것이었습니다. 금전적인 이익, 홍보 효과 등을 노리고 한 것은 전혀 아니었습니다. (앞으로도 계속 그러면 좋겠네요.)

## 저의 결론
제가 이 앱을 공개하고 계속 배포한 이유는, 사람들에게 주는 편익이 크고, 자가진단의 실효성이 작다고 생각했기 때문입니다. 저는 사람들의 삶을 편하게 하는 개발자가 되고 싶고, 그것에 대한 한 가지 방법이라고 생각했습니다.
... 그런데 요즘은 회의감이 들기도 하네요. 전 우유부단한 사람이라 내리고 싶다는 것으로 마음이 기우네요. 그냥 내릴까요

## 결론
여기서는 제가 모든 의견을 제시한 것이 아니라, 그냥 제가 고민해보았던 것을 글로 옮겼을 뿐입니다. 제가 이런 생각을 했다는 것을 보여주는 것이죠. 사실 모든 생각이 한번에 떠오르지는 않아서 주기적으로 새로운 생각이 떠오를 때마다 업데이트될 겁니다. ~~그리고 업데이트되지 않았다~~  
여러분들의 생각은 어떠신가요?

**Comment by concertypin** [comment]
제가 알기론 매크로는 리버싱이 아니라고 알고 있어요. 저작권법상 기술적 보호조치는 이래요.
```28. "기술적 보호조치"란 다음 각 목의 어느 하나에 해당하는 조치를 말한다.
가. 저작권, 그 밖에 이 법에 따라 보호되는 권리의 행사와 관련하여 이 법에 따라 보호되는 저작물등에 대한 접근을 효과적으로 방지하거나 억제하기 위하여 그 권리자나 권리자의 동의를 받은 자가 적용하는 기술적 조치
나. 저작권, 그 밖에 이 법에 따라 보호되는 권리에 대한 침해 행위를 효과적으로 방지하거나 억제하기 위하여 그 권리자나 권리자의 동의를 받은 자가 적용하는 기술적 조치
```
이걸 잘 보면, 매크로는 자가진단의 침해 행위가 있는 것도 아니고 접근을 통제하는 것도 아니죠. 누구든지 쓸 수 있고, 자가진단이 하는 게 그냥 질문 몇 개 받아서 등교 가능/불가 여부만 서버로 넘기는 거니까요.

**Reply by lhwdev** [reply]
오오오 법알못이기도 하고 시간이 없어서 못 찾아봤는데.. 감사합니다!
